### PR TITLE
[doc] Contribute a shape for the container border style customization.

### PR DESCRIPTION
**Pull request by frouene**
Shape for https://github.com/eclipse-sirius/sirius-components/issues/2077.

### CODE REVIEW

File: doc/iterations/2023.8/shapes/customize_containers_with_border.adoc [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+** Group
+** Flexbox
+* Reuse of existing `BorderStyle` in the view DSL, this style declares 4 attributs :
+** borderColor
+** borderRadius
+** borderSize
+** borderLineStyle
+* For a container, the `borderStyle` is optional and doesn't exist by default.
+* There can be only one `borderStyle`, but several conditional styles.
+* The conditional border style condition property is an expression that will be computed in the context of the container's semantic element.
```
Comment (pcdavid) [inline doc/iterations/2023.8/shapes/customize_containers_with_border.adoc]: Not sure why these points are "rabbit holes".

File: doc/iterations/2023.8/shapes/customize_containers_with_border.adoc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+=== Cutting backs
+
+Do not offer the possibility of declaring conditional styles on these containers.
+
+== Rabbit holes
+
+* There are two types of container in the View DSL :
+** Group
+** Flexbox
+* Reuse of existing `BorderStyle` in the view DSL, this style declares 4 attributs :
```
Comment (pcdavid) [inline doc/iterations/2023.8/shapes/customize_containers_with_border.adoc]: Typo: "attributs" => "attribut[e]s"

**Review by pcdavid — DISMISSED** [review]

File: doc/iterations/2023.8/shapes/customize_containers_with_border.adoc (hunk 1 above)
Comment (sbegaudeau) [reply doc/iterations/2023.8/shapes/customize_containers_with_border.adoc]: I would simply add that by reusing `BorderStyle` which has been created for the diagram DSL, you are exposing the view DSL to some minor risks. If / when we will want to change things for one of those DSLs, we will have to change it for everybody (for example, when we will allow specifier to indicate different size, color, style of border for the top, left, bottom, right borders).

I'm not against this but we should just have it in mind, especially when we will split the view DSL in X parts.

Comment (frouene) [reply doc/iterations/2023.8/shapes/customize_containers_with_border.adoc]: I moved most of these points to the solution part.

Comment (frouene) [reply doc/iterations/2023.8/shapes/customize_containers_with_border.adoc]: I've specified that a change to `BorderStyle` will affected all container border style.

File: doc/iterations/2023.8/shapes/customize_containers_with_border.adoc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+In forms, it is difficult to identify the limits of a container (group or flexbox).
+
+== Key Result
+
+Give a specifier the possibility to add to each container a style with border.
+
+== Solution
+
+In the View DSL, creation of a new style named `ContainerBorderStyle` with 4 attributes:
```
Comment (pcdavid) [inline doc/iterations/2023.8/shapes/customize_containers_with_border.adoc]: Now in the **Form** DSL.

**Review by AxelRICHARD — APPROVED** [review]

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**